sdk/storage/Azure.Storage.Blobs/tests/BlobTestBase.cs

```diff
@@ -67,6 +75,49 @@ private BlobServiceClient GetServiceClientFromSharedKeyConfig(TenantConfiguratio
                     new Uri(config.BlobServiceEndpoint),
                     new StorageSharedKeyCredential(config.AccountName, config.AccountKey),
                     this.GetOptions()));
+        
+             
+        private BlobServiceClient GetSecondaryStorageReadEnabledServiceClient(TenantConfiguration config, int numberOfReadFailuresToSimulate, out TestExceptionPolicy testExceptionPolicy, bool simulate404 = false, List<RequestMethod> enabledRequestMethods = null)
+        {
+            var options = this.getSecondaryStorageOptions(config, out testExceptionPolicy, numberOfReadFailuresToSimulate, simulate404, enabledRequestMethods);
+            return this.InstrumentClient(
+                 new BlobServiceClient(
+                    new Uri(config.BlobServiceEndpoint),
+                    new StorageSharedKeyCredential(config.AccountName, config.AccountKey),
+                    options));
+        }
+
+        private BlobBaseClient GetSecondaryStorageReadEnabledBlobBaseClient(TenantConfiguration config, int numberOfReadFailuresToSimulate, out TestExceptionPolicy testExceptionPolicy, bool simulate404 = false, List<RequestMethod> enabledRequestMethods = null)
+        {
+            var options = this.getSecondaryStorageOptions(config, out testExceptionPolicy, numberOfReadFailuresToSimulate, simulate404, enabledRequestMethods);
+            return this.InstrumentClient(
+                 new BlobBaseClient(
+                    new Uri(config.BlobServiceEndpoint),
+                    new StorageSharedKeyCredential(config.AccountName, config.AccountKey),
+                    options));
+        }
+
+        private BlobContainerClient GetSecondaryStorageReadEnabledBlobContainerClient(TenantConfiguration config, int numberOfReadFailuresToSimulate, out TestExceptionPolicy testExceptionPolicy, bool simulate404 = false, List<RequestMethod> enabledRequestMethods = null)
+        {
+            var options = this.getSecondaryStorageOptions(config, out testExceptionPolicy, numberOfReadFailuresToSimulate, simulate404, enabledRequestMethods);
+            Uri uri = new Uri(config.BlobServiceEndpoint);
+            string containerName = this.GetNewContainerName();
+            return this.InstrumentClient(
+                 new BlobContainerClient(
+                    uri.AppendToPath(containerName),
+                    new StorageSharedKeyCredential(config.AccountName, config.AccountKey),
+                    options));
+        }
+
+        private BlobClientOptions getSecondaryStorageOptions(TenantConfiguration config, out TestExceptionPolicy testExceptionPolicy, int numberOfReadFailuresToSimulate = 1, bool simulate404 = false, List<RequestMethod> enabledRequestMethods = null)
+        {
+            var options = this.GetOptions();
+            options.GeoRedundantSecondaryUri = new Uri(config.BlobServiceSecondaryEndpoint);
+            options.Retry.MaxRetries = 4;
```

Comment:
Nit - maybe don't set this so we're covered if we change the definition of `GetOptions`?

Reply:
By setting it here we don't have to worry about the tests breaking if the value used in GetOptions changes.